spark_pipeline_framework/transformers/framework_fill_na_transformer/v1/framework_fill_na_transformer.py

```diff
@@ -0,0 +1,118 @@
+from spark_pipeline_framework.progress_logger.progress_log_metric import (
+    ProgressLogMetric,
+)
+from spark_pipeline_framework.progress_logger.progress_logger import ProgressLogger
+from spark_pipeline_framework.transformers.framework_transformer.v1.framework_transformer import (
+    FrameworkTransformer,
+)
+
+
+def get_dtype(df: DataFrame, colnames: List[str]) -> Dict[str, str]:
+    return dict(df.select(colnames).dtypes)
+
+
+class FrameworkFillNaTransformer(FrameworkTransformer):
+    """
+    Replace NA/Null values with a specified replacement_value for specified columns_to_check for null values
+    """
+
+    # noinspection PyUnusedLocal
+    @keyword_only
+    def __init__(
+        self,
+        view: str,
+        replacement_value: Any,
+        columns_to_check: List[str],
+        name: Optional[str] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        progress_logger: Optional[ProgressLogger] = None,
+    ):
+        super().__init__()
+        self.logger = get_logger(__name__)
+
+        self.view: Param = Param(self, "view", "")
+        self._setDefault(view=view)
+
+        self.replacement_value: Param = Param(self, "replacement_value", "")
+        self._setDefault(replacement_value=replacement_value)
+
+        self.columns_to_check: Param = Param(self, "columns_to_check", "")
+        self._setDefault(columns_to_check=columns_to_check)
+
+        kwargs = self._input_kwargs
+        self.setParams(**kwargs)
+
+    @keyword_only
+    def setParams(
+        self,
+        view: str,
+        replacement_value: Any,
+        columns_to_check: List[str],
+        name: Optional[str] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        progress_logger: Optional[ProgressLogger] = None,
+    ) -> Any:
+        kwargs = self._input_kwargs
+        super().setParams(
+            name=name, parameters=parameters, progress_logger=progress_logger
+        )
+        return self._set(**kwargs)
+
+    def _transform(self, df: DataFrame) -> DataFrame:
+        columns_to_fill: List[str] = self.getColumnsToCheck()
+        replacement_value: Any = self.getReplacementValue()
+        view: str = self.getView()
+        progress_logger: Optional[ProgressLogger] = self.getProgressLogger()
+
+        with ProgressLogMetric(name=f"{view}_fill_na", progress_logger=progress_logger):
+            self.logger.info(
+                f"filling rows if any null values with replacement_value found for columns: {columns_to_fill}"
+            )
+            df_with_na: DataFrame = df.sql_ctx.table(view)
+            data_types = get_dtype(df_with_na, columns_to_fill)
+
+            if type(replacement_value) in (list, dict):
+                assert len(replacement_value) == len(
+                    columns_to_fill
+                ), f"If replacement_value is a list or dictionary, the values must be equal to the number of columns in columns to fill. {len(replacement_value)} != {len(columns_to_fill)}"
```

Comment:
I think it makes more sense to me, seeing this, to use a dict or kwargs to define the fill values for each column.

something like:

```
FrameworkFillNaTransformer(
        view=view,
        column_mapping={
             "Minimum Age": 1.0,
             "Maximum Age": 99.0,
             "Something Else": "N/A",
        },
    ).transform(df)
```

Reply:
Are you saying it should exclusively use a dictionary?

Reply:
I'm changing it, cause your suggestion makes too much sense to me not to do it.

Reply:
If the options are either a pair of lists matched by position or a dictionary, then yes, I absolutely would say to exclusively use a dictionary.

```
FrameworkFillNaTransformer(
        view=view,
        column_mapping={"Minimum Age": 1.0,},
    ).transform(df)
```

is much cleaner than
```
FrameworkFillNaTransformer(
        view=view,
        columns=[
             "Minimum Age",
             "Maximum Age",
             "Somethine Else",
        ],
        fill_value=[
             1.0,
             99.0,
             "N/A",
        ],
    ).transform(df)
```

or even
```
FrameworkFillNaTransformer(
        view=view,
        columns=[
             "Minimum Age",
        ],
        fill_value=1.0,
    ).transform(df)
```

If you have an alternative though, I'm open.